### PR TITLE
docs: document format of locale codes, allFakers and allLocales

### LINKED ISSUE #1909: faker.allLocales/faker.allFakers is undocumented
Followon from https://github.com/faker-js/faker/pull/1735

This should be documented somewhere, e.g. on the Localization page?

### PR DESCRIPTION
**Pull request by matthewmayer**
extracted from #2025
fixes #1909

### CODE REVIEW

**Comment by ST-DDT** [comment]
There seems to be some kind of prettier formatting issue. Could you please check whats wrong and fix it?

File: docs/guide/localization.md [hunk 1]
```diff
@@ -144,3 +144,19 @@ The `Locale` (data) and `Faker` columns refer to the respective `import` names:
...
+
+Rarely, an additional variant may be needed to fully represent a accented version of the locale, or for languages which can be written in different scripts. This is appended after another underscore, for example `en_AU_ocker` (English in Australia in "Ocker" dialect) or `sr_RS_latin` (Serbian in Serbia in Latin script).
+
+You can access all prebuilt Faker instances or all locale definitions via an object where the locale codes are the keys:
+
+```ts
+import { allFakers, allLocales } from '@faker-js/faker';
+console.dir(allFakers['de_AT']); // the prebuilt Faker instance for de_AT
+console.dir(allLocales['de_AT']); // the raw locale definitions for de_AT
+```
```
Comment (ST-DDT) [inline docs/guide/localization.md:162]: I would add a hint that we recommend individual imports instead.

**Review by ST-DDT — COMMENTED** [review]
Looks good to me. Thanks!

File: docs/guide/localization.md (hunk 1 above)
Comment (matthewmayer) [reply docs/guide/localization.md:162]: added note and further example

**Review by ST-DDT — DISMISSED** [review]
I'm not sure about the enumerating usecase, but otherwise this looks good to me.

**Review by xDivisionByZerox — DISMISSED** [review]

File: docs/guide/localization.md [hunk 2]
```diff
@@ -144,3 +144,34 @@ The `Locale` (data) and `Faker` columns refer to the respective `import` names:
...
 import { de, fakerDE } from '@faker-js/faker';
 ```
+
+## Locale codes
+
+Locales are named in a systematic way. The first two characters are a lowercase language code following the [ISO 639-1 standard](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) for example `ar` for Arabic or `en` for English.
+
+The same language may be spoken in different countries, with different patterns for addresses, phone numbers etc. Optionally a two-letter uppercase country code can be added after an underscore, following the [ISO 3166-1 alpha-2 standard](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2), for example `en_US` represents English (United States) and `en_AU` represents English (Australia).
+
+Rarely, an additional variant may be needed to fully represent a accented version of the locale, or for languages which can be written in different scripts. This is appended after another underscore, for example `en_AU_ocker` (English in Australia in "Ocker" dialect) or `sr_RS_latin` (Serbian in Serbia in Latin script).
```
Comment (Shinigami92) [inline docs/guide/localization.md]:
```suggestion
Rarely, an additional variant may be needed to fully represent an accented variant of the locale, or for languages which can be written in different scripts. This is appended after another underscore, for example `en_AU_ocker` (English in Australia in "Ocker" dialect) or `sr_RS_latin` (Serbian in Serbia in Latin script).
```

**Review by Shinigami92 — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**